docs/data/introduction/licensing/licensing.md

```diff
@@ -195,16 +195,29 @@ export default function RootLayout(props: { children: React.ReactNode }) {
 
 ### Next.js Pages Router
 
-When using Next.js pages, a great place to call `setLicenseKey` is in [`_app.js`](https://nextjs.org/docs/pages/building-your-application/routing/custom-app).
+When using Next.js pages, a great place to call `setLicenseKey` is in [`_app.tsx`](https://nextjs.org/docs/pages/building-your-application/routing/custom-app).
 
-```tsx
+```tsx title="pages/_app.tsx"
+import type { ReactElement, ReactNode } from 'react';
+import type { NextPage } from 'next';
+import type { AppProps } from 'next/app';
 import { LicenseInfo } from '@mui/x-license';
 
 LicenseInfo.setLicenseKey('YOUR_LICENSE_KEY');
 
-export default function MyApp(props) {
-  const { Component, pageProps } = props;
-  return <Component {...pageProps} />;
+export type NextPageWithLayout<P = {}, IP = P> = NextPage<P, IP> & {
+  getLayout?: (page: ReactElement) => ReactNode;
+};
+
+type AppPropsWithLayout = AppProps & {
+  Component: NextPageWithLayout;
+};
+
+export default function MyApp({ Component, pageProps }: AppPropsWithLayout) {
```

Comment:
I have been constantly normalizing towards having all our code to not destrructre the props in the argument side, to make it super clear that this is a component.

```suggestion
export default function MyApp(props: AppPropsWithLayout) {
  const { Component, pageProps } = props;
```